packages/block-editor/src/components/link-control/search-item.js

```diff
@@ -9,7 +9,44 @@ import classnames from 'classnames';
 import { safeDecodeURI, filterURLForDisplay } from '@wordpress/url';
 import { __ } from '@wordpress/i18n';
 import { Button, TextHighlight } from '@wordpress/components';
-import { Icon, globe } from '@wordpress/icons';
+import {
+	Icon,
+	globe,
+	page,
+	tag,
+	postList,
+	category,
+	file,
+} from '@wordpress/icons';
+
+function SearchItemIcon( { isURL, suggestion } ) {
+	let icon = null;
+
+	if ( isURL ) {
+		icon = globe;
+	} else if ( suggestion.type === 'post' ) {
+		icon = postList;
+	} else if ( suggestion.type === 'page' ) {
+		icon = page;
+	} else if ( suggestion.type === 'post_tag' ) {
+		icon = tag;
+	} else if ( suggestion.type === 'category' ) {
+		icon = category;
+	} else if ( suggestion.type === 'attachment' ) {
+		icon = file;
+	}
```

Comment:
Nit: would you consider having a map of suggestion types to icons and then just look up by key?

It would be easier to maintain than a list of `if/else`.

Reply:
Sure thing!